django_mongodb/client.py

```diff
@@ -4,11 +4,51 @@
 
 
 class DatabaseClient(BaseDatabaseClient):
-    executable_name = "mongo"
+    executable_name = "mongosh"
+    help_options = ("--help", "-h")
 
     @classmethod
     def settings_to_cmd_args_env(cls, settings_dict, parameters):
-        raise NotImplementedError
+        options = settings_dict.get("OPTIONS", {})
+        uri = options.get("uri")
+        args = [cls.executable_name]
+        if not uri:
+            host = settings_dict["HOST"]
+            port = settings_dict["PORT"]
+            if port and host:
+                host += f":{port}"
+
+            dbname = settings_dict["NAME"]
+            user = settings_dict["USER"]
+            passwd = settings_dict["PASSWORD"]
+            auth_database = options.get("authenticationDatabase")
+            auth_mechanism = options.get("authenticationMechanism")
+            retry_writes = options.get("retryWrites")
+            protocol = options.get("protocol", "mongodb")
+
+            if user:
+                args += ["--username", user]
+            if passwd:
+                args += ["--password", passwd]
+            if auth_database:
+                args += ["--authenticationDatabase", auth_database]
+            if auth_mechanism:
+                args += ["--authenticationMechanism", auth_mechanism]
+            if retry_writes is not None:
+                args += ["--retryWrites", str(retry_writes).lower()]
+            if host:
+                uri = f"{protocol}://{host}/{dbname}"
+            elif dbname:
+                uri = dbname
+
+        if uri:
+            args.append(uri)
+
+        if parameters:
+            if not any(param in cls.help_options for param in parameters):
```

Comment:
I'm not sure what the purpose of this is. It looks like it could be to try to invoke `mongosh --help`. I don't think it works.  `python manage.py dbshell --help` prints the help info for dbshell. Can `--help` make it to `parameters`? Anyway, this is more complicated than it needs to be. Just keep things minimal for now.

Reply:
The parameters could be the scripts files, the client could use as pre-run scripts. I don't know if it is massively used, maybe @Jibola could have a better opinion.
Other thing is: the help is mongosh help. I thought the help should display the help from mongosh, I need to double check what it does with psql (postgres).

Reply:
Did you test `manage.py dbshell --help`? It prints dbshell help as far as I can see. I don't think this code can be reached. If it works, please explain how to invoke it.

Reply:
is there no way to ensure things pipe properly? 
Something like 
`manage.py dbshell -- --help` rather than `manage.py dbshell --help`? I don't think it'll be possible to override the inner `help` command.

Reply:
:thinking:  Yes, it is unreachable. I added this inspired by MySQL unit test. 
https://github.com/mongodb-forks/django/blob/main/tests/dbshell/test_mysql.py#L187